src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,132 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  // eslint-disable-next-line max-len, prettier/prettier
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  const initialInputsState = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [inputs, setInputs] = useState(initialInputsState);
+
+  const isError = // eslint-disable-line
+    !inputs.title || // eslint-disable-line
+    !inputs.imgUrl || // eslint-disable-line
+    !inputs.imdbUrl || // eslint-disable-line
+    !inputs.imdbId; // eslint-disable-line
+
+  const [isImgUrlValid, setIsImgUrlValid] = useState(false);
+  const [isImdbUrlValid, setIsImdbUrlValid] = useState(false);
+  // State below is so that the error about the link appears only after the second
+  // attempt to send the form, so as not to immediately scare the user
+  const [hasUrlError, setIsUrlError] = useState(false);
```

Comment:
Your setter needs to be identical to value name 
```suggestion
  const [hasUrlError, setHasUrlError] = useState(false);
```